akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcProjectionImpl.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+import java.util.concurrent.CompletionStage
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.jdk.FutureConverters._
+
+import akka.Done
+import akka.NotUsed
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.event.Logging
+import akka.projection.Projection
+import akka.projection.ProjectionId
+import akka.projection.ProjectionSettings
+import akka.projection.RunningProjection
+import akka.projection.internal.HandlerLifecycleAdapter
+import akka.projection.jdbc.javadsl.JdbcHandler
+import akka.projection.jdbc.javadsl.JdbcProjection
+import akka.projection.jdbc.javadsl.JdbcSession
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.KillSwitches
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.Source
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[projection] class JdbcProjectionImpl[Offset, Envelope, S <: JdbcSession](
```

Comment:
This is very incomplete, don't bother to look at it for now. 

I started here and then move to the offset store because it's where all the JDBC 'magic' happends.